### PR TITLE
Refactor constants usage to use deconstructors - Closes #2381

### LINKED ISSUE #2381: Prefer deconstruction when using Lisk Core constants
### Expected behavior
We should deconstruct constants object on top of modules like below:

```js
const {EPOC_TIME} = global.constants;
```
### Actual behavior
We are trying to access the constants via constants variable.

```js
const constants = global.constants;

// ...

console.log(constants.EPOCH_TIME)
```

See Discussion here: https://github.com/LiskHQ/lisk/pull/2371#discussion_r215214501
### Which version(s) does this affect? (Environment, OS, etc...)
1.3.0+

### PR DESCRIPTION
**Pull request by yatki**
### Review checklist

* The PR resolves #2381 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: logic/account.js [hunk 1]
```diff
@@ -306,7 +306,7 @@ class Account {
 			}
 		});
 
-		let limit = constants.ACTIVE_DELEGATES;
+		let limit = ACTIVE_DELEGATES;
```
Comment (ManuGowda) [inline logic/account.js]: I don't see any usage of limit in this file, can you remove it if its not being used.

Comment (yatki) [reply logic/account.js]: @ManuGowda it is being used [here](https://github.com/LiskHQ/lisk/blob/2381-prefer-deconstruction/logic/account.js#L344)

File: helpers/slots.js [hunk 2]
```diff
@@ -38,15 +38,12 @@ function getEpochTime(time) {
...
-	var t = d.getTime();
-
-	return Math.floor((time - t) / 1000);
+	return Math.floor((time - EPOCH_TIME.getTime()) / 1000);
 }
 
 module.exports = {
 	interval: 10,
-	delegates: constants.ACTIVE_DELEGATES,
+	delegates: ACTIVE_DELEGATES,
```
Comment (4miners) [inline helpers/slots.js]: We shouldn't export `ACTIVE_DELEGATES` as `delegates` here and use `ACTIVE_DELEGATES` in corresponding modules that uses `slots.delegates`, can we fix this?

File: helpers/config.js [hunk 3]
```diff
@@ -283,13 +283,14 @@ function cleanDeep(
  * @param {Object} configData
  * @todo Add description for the params
  */
-function validateForce(configData) {
-	if (configData.forging.force) {
-		var index = configData.constants.NETHASHES.indexOf(configData.nethash);
+function validateForce({ constants, forging, nethash }) {
+	if (forging.force) {
+		const { NETHASHES } = constants;
+		var index = NETHASHES.indexOf(nethash);
```
Comment (4miners) [inline helpers/config.js]: Please use more meaningful variable name.

File: helpers/slots.js [hunk 4]
```diff
@@ -71,8 +68,7 @@ module.exports = {
 			epochTime = this.getTime();
 		}
 
-		var d = constants.EPOCH_TIME;
-		var t = Math.floor(d.getTime() / 1000) * 1000;
+		var t = Math.floor(EPOCH_TIME.getTime() / 1000) * 1000;
```
Comment (4miners) [inline helpers/slots.js]: Same here, would be nice to rename this variable.

File: logic/block_reward.js [hunk 5]
```diff
@@ -33,13 +33,13 @@ const __private = {};
 class BlockReward {
 	constructor() {
 		// Array of milestones
-		this.milestones = constants.REWARDS.MILESTONES;
+		this.milestones = REWARDS.MILESTONES;
 
 		// Distance between each milestone
-		this.distance = Math.floor(constants.REWARDS.DISTANCE);
+		this.distance = Math.floor(REWARDS.DISTANCE);
```
Comment (4miners) [inline logic/block_reward.js]: Why we need to call `Math.floor` on this?

File: logic/block_reward.js [hunk 6]
```diff
@@ -33,13 +33,13 @@ const __private = {};
...
 
 		// Start rewards at block (n)
-		this.rewardOffset = Math.floor(constants.REWARDS.OFFSET);
+		this.rewardOffset = Math.floor(REWARDS.OFFSET);
```
Comment (4miners) [inline logic/block_reward.js]: Same here.

File: logic/block_reward.js [hunk 7]
```diff
@@ -33,13 +33,13 @@ const __private = {};
 class BlockReward {
 	constructor() {
 		// Array of milestones
-		this.milestones = constants.REWARDS.MILESTONES;
+		this.milestones = REWARDS.MILESTONES;
```
Comment (4miners) [inline logic/block_reward.js]: Why do we need to assign those instead of using directly?

File: helpers/config.js (hunk 3 above)
Comment (yatki) [reply helpers/config.js]: I removed the variable since we didn't need it.

File: helpers/slots.js (hunk 4 above)
Comment (yatki) [reply helpers/slots.js]: I removed the variable here as well

File: logic/block_reward.js (hunk 5 above)
Comment (yatki) [reply logic/block_reward.js]: @4miners I can't comment on this. Is there any chance that rewards constants can be float?

File: logic/block_reward.js (hunk 7 above)
Comment (yatki) [reply logic/block_reward.js]: Removed: `this.milestones`.

File: logic/block_reward.js (hunk 5 above)
Comment (yatki) [reply logic/block_reward.js]: @ManuGowda can you comment on this ?

Comment (4miners) [reply logic/block_reward.js]: They cannot.

Comment (yatki) [reply logic/block_reward.js]: @4miners then having `this.distance` variable doesn't make sense either.

Comment (4miners) [reply logic/block_reward.js]: Yep, I was referencing to all of them.

File: logic/account.js (hunk 1 above)
Comment (ManuGowda) [reply logic/account.js]: ok, can change let to const its not being reassigned.

File: logic/block_reward.js (hunk 5 above)
Comment (ManuGowda) [reply logic/block_reward.js]: you can directly use `REWARDS.DISTANCE` directly. I don't know the intention behind `floor` logic here, but as Mariusz said rewards will not be a float.

Comment (yatki) [reply logic/block_reward.js]: @4miners Let's handle this in a separate issue. see: https://github.com/LiskHQ/lisk/issues/2412

File: logic/block_reward.js (hunk 6 above)
Comment (yatki) [reply logic/block_reward.js]: same above.

File: helpers/slots.js (hunk 2 above)
Comment (yatki) [reply helpers/slots.js]: this requires a big refactor. Let's do it in a separate issue. See: https://github.com/LiskHQ/lisk/issues/2413

Comment (yatki) [reply helpers/slots.js]: @4miners see my comment above.

Comment (4miners) [reply helpers/slots.js]: Ahh, ok, sorry. 🤗

File: logic/account.js (hunk 1 above)
Comment (ManuGowda) [reply logic/account.js]: @yatki did you address the limit to `const`?

Comment (yatki) [reply logic/account.js]: @ManuGowda it is being reassigned [here](https://github.com/LiskHQ/lisk/blob/2381-prefer-deconstruction/logic/account.js#L319)

**Pull request merged**